.github/workflows/upload.yml

```diff
@@ -0,0 +1,163 @@
+
+      - name: Install KaHyPar
+        run: pip3 install kahypar==1.1.7
+
+      - name: Install PennyLane
+        run: |
+          pip install -r requirements-ci.txt
+          python setup.py bdist_wheel
+          pip install dist/PennyLane*.whl
+
+      - name: Install Lightning-master
+        run: python -m pip install -i https://test.pypi.org/simple/ PennyLane-Lightning --pre --upgrade
+
+      - name: Run tests
+        run: python -m pytest tests --cov=pennylane $COVERAGE_FLAGS
+
+  device-tests:
+    runs-on: ubuntu-latest
+
+    strategy:
+      matrix:
+        config:
+          - { device: "default.qubit", shots: None }
+          - { device: "default.qubit", shots: 10000 }
+          # - {device: "default.qubit.tf", shots: None}
+          - { device: "default.qubit.autograd", shots: None }
+          - { device: "default.mixed", shots: None }
+
+    steps:
+      - name: Cancel Previous Runs
+        uses: styfle/cancel-workflow-action@0.4.1
+        with:
+          access_token: ${{ github.token }}
+
+      - uses: actions/checkout@v2
+        with:
+          fetch-depth: 2
+
+      - name: Set up Python
+        uses: actions/setup-python@v2
+        with:
+          python-version: 3.8
+
+      - name: Install dependencies
+        run: |
+          python -m pip install --upgrade pip
+          pip install -r requirements-ci.txt
+          pip install wheel pytest pytest-cov pytest-mock flaky --upgrade
+
+      - name: Conditionally install PyTorch
+        if: contains(matrix.config.device, 'torch')
+        run: pip3 install torch==$TORCH_VERSION -f https://download.pytorch.org/whl/torch_stable.html
+
+      - name: Conditionally install TensorFlow
+        if: contains(matrix.config.device, 'tf')
+        run: pip3 install tensorflow==$TF_VERSION
+
+      - name: Conditionally install Jax
+        if: contains(matrix.config.device, 'jax')
+        run: pip3 install jax jaxlib
+
+      - name: Install PennyLane
+        run: |
+          pip install -r requirements-ci.txt
+          python setup.py bdist_wheel
+          pip install dist/PennyLane*.whl
+
+      - name: Run tests
+        run: |
+          python -m pytest pennylane/devices/tests \
+            --device=${{ matrix.config.device }} \
+            --shots=${{ matrix.config.shots }} \
+            --cov=pennylane $COVERAGE_FLAGS
+
+
+  upload:
+    runs-on: ubuntu-latest
+    needs: [core-tests, device-tests]
+    if: ${{ always() && !cancelled() && needs.build.result == 'success' }}
+    steps:
+      - uses: actions/checkout@v2
+
+      - name: Set up Python
+        uses: actions/setup-python@v2
+        with:
+          python-version: 3.8
+
+      - name: Build and install PennyLane
+        run: |
+          python -m pip install --upgrade pip wheel
+          pip install -r requirements-ci.txt
+          python setup.py bdist_wheel
+          pip install dist/PennyLane*.whl
+
+      - name: Publish
+        uses: pypa/gh-action-pypi-publish@master
```

Comment:
@antalszava Should it be the latest rc branch?

Reply:
This refers to the branch of the github action no?

Reply:
It is where I am not sure at all, it is done like that in the plugins. But here with the release candidate branch, I was not sure.

Reply:
Yeah so this line instructs to use the master branch version of the `gh-action-pypi-publish` github action, which is fine I guess. Normally you would probably use a release version here like with other actions (e.g. checkout@v2)